### PR TITLE
feat: tangle avs

### PR DESCRIPTION
**Pull request by Tjemmmic**

### CODE REVIEW

**Comment by Tjemmmic** [comment]
Most of the functionality is there, I am just debugging the last few things in getting the whole flow to work in the test. This PR should be good to merge in the next day.

File: build.rs [hunk 1]
```diff
@@ -1,4 +1,58 @@
+use std::env;
```
Comment (drewstone) [inline build.rs:1]: We're using this quite a lot we may want to have a `blueprint-build-utils` crate in `gadget` that can just expose this for different contract_dirs and be reused.

File: src/utils/eigenlayer.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/// # Environment Variables
+/// - `EIGENLAYER_HTTP_ENDPOINT`: HTTP endpoint for EigenLayer.
+/// - `EIGENLAYER_WS_ENDPOINT`: WebSocket endpoint for EigenLayer.
+///
+/// # Errors
+/// May return errors if:
+/// - Environment variables are not set.
+/// - Contract interactions fail.
+/// - Registration processes encounter issues.
+pub async fn register_to_eigenlayer(
```
Comment (drewstone) [inline src/utils/eigenlayer.rs:28]: Presumably this goes away once you're using the new `BlueprintRunner` right?

**Review by drewstone — APPROVED** [review]
Looks good to merge

**Pull request merged**